PYTHON/activityLog/activityLog.py

```diff
@@ -0,0 +1,356 @@
+import os
+import sys
+import shutil
+import subprocess
+import platform
+import re
+import argparse
+from datetime import datetime
+import time
+import tarfile
+
+### TODO: redesign as a class object (OOP)
+
+class ActivityLogger(object):
+	"""Generates a log file for server activity, CPU load, and MySQL statistics."""
+	def __init__(self):
+		super(ActivityLogger, self).__init__()
+		self.checkcronjob()
+		self.epochflag = False
+		self.rootcheck = False
+		self.mysql_usr = None
+		self.mysql_pass = None
+		self.mysql_queries = 0
+		self.mysql_interval = 0
+		self.logfilename = None
+		### Get server specs
+		self.current_platform = platform.platform().lower() # platform information in lower case
+
+		
+	def checkcronjob(self):
+		# Checks to see if script was run by user, or a cron job.
+		# Will create actLogDir in home dir if cron, current working directory 
+		# if run by user.
+		# Explained: sys.stdin will be a TTY if run by console (by the user).
+		if os.isatty(sys.stdin.fileno()):
+			self.cronjob = False
+			self.silent_mode = False
+		else:
+			self.cronjob = True
+			self.silent_mode = True
+
+
+	# Returns a tuple of the stdout and stderr of a command. 
+	# NOTE that the use of shell=True is extremely dangerous from a security standpoint. Using it
+	# here because the alternative (parsing each command and argument into a list) makes piping
+	# multiple commands difficult.
+	def run_command(self, cmd):
+		return subprocess.Popen(
+			cmd, 
+			shell=True, 
+			stdout = subprocess.PIPE, # stdout in [0]
+			stderr = subprocess.PIPE).communicate() # stderr in [1]
+
+	def dateString(self, epoch = False):
+		# returns time from epoch.  ON UNIX, it is 0 hours on Jan. 1st, 1970
+		if epoch == True: 
+			return time.time()
+		elif epoch != True:
+		# using UTC (Similar to GMC)
+		# strftime converts the struct_time format into a string. see documentation
+			dateTime = datetime.utcnow().strftime("%mm_%dd_%Yy__at_UTC_%Hh_%Mm_%Ss")
+			return dateTime
+
+
+	def rootFsRwRoStateCheck(self):
+	# Checks to see if root (/) directory is read/write 
+		try:
+			# Same output as running "mount" in BASH 
+			with open("/proc/mounts", "r") as f:
+				for line in f:
+					# Look for line with root (/) dir and "rw" in the line
+					if " / " in line and "rw" in line:
+						return True
+				return False
+		except:
+			if self.silent_mode == "False":
+				print("Opening /proc/mounts failed.  Returning False")
+				return False
+			else:
+				return False
+
+
+	def topRC(self):
+		# Custom ~./toprc config.  I have no idea what is happening here.
+
+		# Expands "~" to user home dir
+		toprc_loc = os.path.expanduser("~/.toprc")
+		toprcexists = os.path.isfile(toprc_loc)
+		toprc = """RCfile for \"top with windows\"
```

Comment:
I would usually put stuff like this (big data strings that are basically separate files) at the top or bottom of your file, in the global context.  That way you can put a newline after the opening `"""` and have the content formatted up against the left margin so you don't have to fight your Python indenter all the time.